src/clj/puppetlabs/services/ca/certificate_authority_core.clj

```diff
@@ -329,9 +329,18 @@
 
   :handle-ok
   (fn [context]
-    (->
-      (ca/get-certificate-statuses settings)
-      (as-json-or-pson context))))
+    (let [query-params (:params request)
+          queried-state (get query-params "state")]
+      (->
+        (case queried-state
+          "requested" (ca/filter-requested-certificates settings)
+          "signed" (ca/filter-signed-certificates settings)
+          "revoked" (ca/filter-revoked-certificates settings)
```

Comment:
I've been going back and forth all day about having separate functions for these... Since we are pretty sure the set of filters won't change (there aren't any other valid cert states that I know of), it's probably alright, but this kind of does seem like a lot of boiler plate for what could instead just be something like:
```
(if (some %(= queried-state) ["requested" "signed" "revoked"])
  (ca/filter-by-certificate-state settings queried-state)
  (ca/get-certificate-statuses settings))
```
Thoughts? I don't have strong opinions, there's just not really any processing in the specific methods so I'm not sure it's worth having that layer.

Reply:
that's fair, I like yours.